kv/kvserver/allocator/allocatorimpl: TestAllocatorFullDisks failed

kv/kvserver/allocator/allocatorimpl.TestAllocatorFullDisks [failed](https://tanzanite.cluster.engflow.com/invocation/9365bf6d-6c48-488c-9efb-4ee3d3561699) on master @ [4d045594e8c65b56c82fcf2a1f14ee30cecfef3d](https://github.com/cockroachdb/cockroach/commits/4d045594e8c65b56c82fcf2a1f14ee30cecfef3d):


```
=== RUN   TestAllocatorFullDisks
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestAllocatorFullDisks1789713329
    test_log_scope.go:81: use -show-logs to present logs inline
    allocator_test.go:8536: testStore 7 ran out of space during generation 54 (rangesAdded=1092/1180): disk (capacity=1.0 GiB, available=-16 MiB, used=1.0 GiB, logicalBytes=1.0 GiB), ranges=65, leases=0, queries=0.00, writes=0.00, ioThreshold={{0 0 0 0 0 0}} bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
    allocator_test.go:8598: -- test log scope end --
test logs left over in: outputs.zip/logTestAllocatorFullDisks1789713329
--- FAIL: TestAllocatorFullDisks (1.11s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=6</code>
, <code>shard=1</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAllocatorFullDisks.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33181

--- a/pkg/kv/kvserver/allocator/allocatorimpl/allocator_test.go
+++ b/pkg/kv/kvserver/allocator/allocatorimpl/allocator_test.go
@@ -8524,8 +8524,8 @@ func TestAllocatorFullDisks(t *testing.T) {
 				continue
 			}
 			ts := &testStores[j]
-			// Add [0,3) ranges to the node, simulating splits and data growth.
-			toAdd := alloc.randGen.Intn(3)
+			// Add [0,2) ranges to the node, simulating splits and data growth.
+			toAdd := alloc.randGen.Intn(2)
 			for k := 0; k < toAdd; k++ {
 				if rangesAdded < rangesToAdd {
 					ts.add(rangeSize, 0)

